patches/net/minecraft/world/item/Item.java.patch

```diff
@@ -28,6 +28,24 @@
      }
  
      @Deprecated
+@@ -104,6 +_,17 @@
+         return this.components;
+     }
+ 
++    /** @deprecated Neo: do not use, use {@link net.neoforged.neoforge.event.ModifyDefaultComponentsEvent the event} instead */
++    @org.jetbrains.annotations.ApiStatus.Internal @Deprecated
++    public void modifyDefaultComponentsFrom(net.minecraft.core.component.DataComponentPatch patch) {
++        if (!net.neoforged.neoforge.internal.RegistrationEvents.canModifyComponents()) throw new IllegalStateException("Default components cannot be modified now!");
++        var builder = DataComponentMap.builder().addAll(components);
++        patch.entrySet().forEach(entry -> builder.set((DataComponentType)entry.getKey(), entry.getValue().orElse(null)));
++        var built = builder.build();
++        Properties.validateComponents(built);
++        components = built;
```

Comment:
We should probably push the modified component map through `Item.Properties.COMPONENT_INTERNER` like vanilla does in `Item.Properties.buildComponents()` for the initial one